### ISSUE TITLE
 HTTP Error 308: PERMANENT REDIRECT when creating robot account

### ISSUE THREAD

**Issue by ybrock** [redhat-cop/quay_configuration#18]
Hello,
We were using `herve4m.quay` successfully to provision organizations with Quay 3.12.5

Now, we have moved to `infra.quay_configuration` (v2.5) and when we try to create Robot Account, we get this error : 

```
TASK [quay-ops : evs-golden | Create Robot Accounts] **************************************************************************************************************
task path: /ansible/playbooks/roles/quay-ops/tasks/org.yaml:11
failed: [localhost] (item=push) => changed=false 
  ansible_loop_var: item
  item: push
  msg: 'Unexpected return code when calling https://quay.apps.oci.domain.com/api/v1/organization/evs-golden/robots/push/federation: HTTP Error 308: PERMANENT REDIRECT'
failed: [localhost] (item=pull) => changed=false 
  ansible_loop_var: item
  item: pull
  msg: 'Unexpected return code when calling https://quay.apps.oci.domain.ch/api/v1/organization/evs-golden/robots/pull/federation: HTTP Error 308: PERMANENT REDIRECT'
```

Here is my ansible code : 
```
- name: "{{ quay_org.name }} | Create Robot Accounts"
  infra.quay_configuration.quay_robot:
    name: "{{ quay_org.name }}+{{ item }}"
    description: "Robot account to {{ item }}"
    state: present
  loop:
    - "push"
    - "pull"
```

I have tried to pass `federations: []` and `append: false`  to see what would happen but I get the same result.
Am I missing something or there is an issue with this ?

Kind regards

**Comment by ybrock** [comment]
I did some more digging and I found out that with version 2.4.0 of `infra.guay_configuration` all is working as expected.

So it seems to be a regression introduced with version 2.5 in relation with the "federation" thing ...

**Comment by herve4m** [comment]
Thank you for reporting this issue. It was indeed related to the new robot account federations feature in Quay 3.13, and it was an oversight on my side for not testing the `quay_robot` module against previous versions of Quay. 
It is now fixed in the 2.5.1 version of the collection, which is available from Ansible Galaxy (https://galaxy.ansible.com/ui/repo/published/infra/quay_configuration/).

For Quay version 3.12 and earlier, ensure that you do not set the `federations` parameter at all, so that the module skips that configuration.
